lib/_stream_readable.js

```diff
@@ -3,14 +3,16 @@
 module.exports = Readable;
 Readable.ReadableState = ReadableState;
 
-const EE = require('events').EventEmitter;
-const Stream = require('stream');
 const Buffer = require('buffer').Buffer;
-const util = require('util');
-const debug = util.debuglog('stream');
+const EE = require('events').EventEmitter;
+const Stream = require('internal/stream');
+const debug = require('internal/debuglog')('stream');
+const inherits = require('internal/inherits');
+
+var Duplex;  // Break recursive dependency with _stream_duplex.js.
```

Comment:
This _seems_ like it should be unnecessary, since `_stream_duplex`'s export is hoisted.